### PR TITLE
PUBDEV-5569: minor SqlManager refactorings

### PR DESCRIPTION
**Pull request by sebhrusen**
small SQLManager refactorings to improve code readability (and also helpful when trying to implement categoricals)

### CODE REVIEW

File: h2o-core/src/main/java/water/fvec/Vec.java [hunk 1]
```diff
@@ -394,7 +394,7 @@ public static Vec makeCon(double x, long len, int log_rows_per_chunk, boolean re
 
   public static Vec makeCon(double x, long len, int log_rows_per_chunk, boolean redistribute, byte type) {
     int chunks0 = (int)Math.max(1,len>>log_rows_per_chunk); // redistribute = false
-    int chunks1 = (int)Math.min( 4 * H2O.NUMCPUS * H2O.CLOUD.size(), len); // redistribute = true
```
Comment (seb-h2o) [inline h2o-core/src/main/java/water/fvec/Vec.java]: there's a bunch of those in current codebase: should I create a ticket to review usages of `H2O.NUMCPUS` and `Runtime.getRuntime().availableProcessors()` ?

**Comment by Pscheidl** [comment]
I am not really sure about name of this PR. The related JIRA issue seems to be completely about something different ( https://0xdata.atlassian.net/browse/PUBDEV-5569 ).

File: h2o-core/src/main/java/water/fvec/Vec.java [hunk 2]
```diff
@@ -394,7 +394,7 @@ public static Vec makeCon(double x, long len, int log_rows_per_chunk, boolean re
...
+    int chunks1 = (int)Math.min( 4 * H2O.ARGS.nthreads * H2O.CLOUD.size(), len); // redistribute = true
```
Comment (Pscheidl) [inline h2o-core/src/main/java/water/fvec/Vec.java]: I am not sure if user should have a way to indirectly influence number of vectors in a chunk by settings number of threads available. We surely use it in SQLManager, where number of connections in the pool is derived from the number of threads available, but using this in a Vec is a **big change** and may affect performance greatly, doesn't it ?'
If this is something to change, I'd do it myself in a separate PR with a proper issue and some performance tests before merging it.

File: h2o-core/src/main/java/water/fvec/Vec.java (hunk 1 above)
Comment (Pscheidl) [reply h2o-core/src/main/java/water/fvec/Vec.java]: Yes, calls to Runtime.getruntime.availableProcessors() should be discarded. Let's go for H2O.NUMCPUS calls everywhere we can. That is a good change :+1:

File: h2o-core/src/main/java/water/parser/ParseSetup.java [hunk 3]
```diff
@@ -298,7 +298,7 @@ public static ParseSetup guessSetup( Key[] fkeys, ParseSetup userSetup ) {
       t._gblSetup._chunk_size = FileVec.DFLT_CHUNK_SIZE;
     } else {
       t._gblSetup._chunk_size = FileVec.calcOptimalChunkSize(t._totalParseSize, t._gblSetup._number_columns, t._maxLineLength,
-              Runtime.getRuntime().availableProcessors(), H2O.getCloudSize(), false /*use new heuristic*/, true);
+              H2O.ARGS.nthreads, H2O.getCloudSize(), false /*use new heuristic*/, true);
```
Comment (Pscheidl) [inline h2o-core/src/main/java/water/parser/ParseSetup.java]: This change could go along with Vec change as well. I'd personally do a little test showing this change does not slow us down in edge cases, provided nthreads are user-defined and can easily be a higher number than H2O.NUMCPUS.

File: h2o-core/src/main/java/water/parser/ParseSetup.java [hunk 4]
```diff
@@ -665,7 +665,7 @@ private static final void checkEncoding(byte[] bits) {
    * @return The longest line length in the given bytes
    */
   private static final int maxLineLength(byte[] bytes) {
-    int start = bytes.length;
+    int start = bytes.length; //??? mistake? why not init at 0?
```
Comment (Pscheidl) [inline h2o-core/src/main/java/water/parser/ParseSetup.java]: Good catch. We should add a test for it.

File: h2o-core/src/main/java/water/jdbc/SQLManager.java [hunk 5]
```diff
@@ -297,7 +340,7 @@ static int getMaxConnectionsPerNode(final int cloudSize, final short nThreads, f
      * @param maxTotalConnections Maximal number of total connections to be opened by the whole cluster
      * @return Number of connections to open per node, within given minmal and maximal range
      */
-    private static final int calculateLocalConnectionCount(final int maxTotalConnections, final int cloudSize,
+    private static int calculateLocalConnectionCount(final int maxTotalConnections, final int cloudSize,
```
Comment (Pscheidl) [inline h2o-core/src/main/java/water/jdbc/SQLManager.java]: Why not non-final ? Not that it matters ... Just a question, nothing really important in our API design I believe.

**Comment by Pscheidl** [comment]
Added my opinions. I find the chages made to be good and desirable, yet we should probably test them better and split the PR (let's discuss).

File: h2o-core/src/main/java/water/fvec/Vec.java (hunk 2 above)
Comment (seb-h2o) [reply h2o-core/src/main/java/water/fvec/Vec.java]: I understand the concern, but the number of chunks is deeply connected with parallelism when manipulating the vector, right? and parallelism is associated with nthreads, not its default NUMCPUS, so why was the non-configurable NUMCPUS chosen here in the first place.
